### ISSUE TITLE
Building with rpaths on OSX High Sierra

### ISSUE THREAD

**Issue by rijobro** [CCPPETMR/SIRF-SuperBuild#63]
With OSX, dynamic libraries can be referred to as (using the example of libboostfilesystem.dylib):
1. `libboostfilesystem.dylib`,
2. `@rpath/libboostfilesystem.dylib`, or
3. `/absolute/path/to/libboostfilesystem.dylib`.

Compilation against the first of these three causes linking errors on my version of OSX. This is possibly due to changes to SIP for High Sierra (or possible a problem specific to my machine). 

Possible solutions include: 
1. using install_name_tool to change the first option to the second or third, or
2. linking against static libraries.

Hopefully this issue will sort itself out over time.

**Comment by KrisThielemans** [comment]
@rijobro, I think this can be closed?

**Comment by rijobro** [comment]
The option, `SHARED_LIBS_ABS_PATH`, means that all dynamic libraries compiled by CMake will have the full paths, and so the problem is resolved. However, for dependencies that are not compiled with CMake (e.g., Boost, which is compiled with boostrap), the problem remains.

A short-term fix is to use `brew install boost` (or `brew upgrade boost`, if they're not using the current version, 1.66).

As for long-term, I'm not so sure. Boost isn't compiled with CMake; what else isn't? We would have to find fixes for all of those...

**Comment by KrisThielemans** [comment]
I was wondering if this wouldn't be fixed if we'd let the SuperBuild use boost 1.66. I saw no reason not to, until I found that [boost 1.66 doesn't compile on Travis OSX](https://github.com/boostorg/build/issues/279), and therefore presumably some versions of OSX. Sigh.

I therefore suggest that on OSX, we currently set the USE_SYSTEM_BOOST to ON by default, and document the `brew` stuff in our OSX installation instructions.
